internal/controller/k8s/k8scluster/cluster.go

```diff
@@ -221,16 +220,19 @@ func (c *externalCluster) Delete(ctx context.Context, mg resource.Managed) error
 	}
 
 	cr.SetConditions(xpv1.Deleting())
-	if cr.Status.AtProvider.State == compute.DESTROYING || cr.Status.AtProvider.State == k8s.TERMINATED {
+	switch cr.Status.AtProvider.State {
+	case k8s.DESTROYING:
 		return nil
-	}
-	if cr.Status.AtProvider.State != compute.ACTIVE {
+	case k8s.TERMINATED:
+		return nil
+	case k8s.ACTIVE:
+		apiResponse, err := c.service.DeleteK8sCluster(ctx, cr.Status.AtProvider.ClusterID)
+		if err != nil {
+			retErr := fmt.Errorf("failed to delete k8s cluster. error: %w", err)
+			return compute.AddAPIResponseInfo(apiResponse, retErr)
```

Comment:
Please use `CheckAPIResponseInfo()` just like it was previously set, in order to check 404 status.